refactor vertex-edge graph and dual graph of a polyhedron

micjoswig opened the pull request:
This is the obvious minimal implementation for vertex edge graphs and dual graphs of polyhedra; taken from polymake.

The question is how to deal with unbounded polyhedra, in particular those with a nontrivial lineality space.  The functions will work in these cases, too.  However, the behavior might be unexpected.  polymake considers the projectivized quotient modulo linealities, which is always a polytope.  Oscar, on the other hand, follows a more naive semantics.

Opinions?

micjoswig committed:
vertex-edge graph and dual graph of a polyhedron; first draft

micjoswig committed:
formatter

micjoswig commented:
And, of course, we need to care of cones, polyhedral complexes etc, too.

benlorenz committed:
Graphs: rename polyhedron graph methods, fix unbounded cases

lkastner commented on src/Combinatorics/Graphs/functions.jl:
```diff
@@ -902,20 +936,21 @@ end
 vertices of the polyhedron, there is an edge between two vertices if the
 polyhedron has an edge between the corresponding vertices. The resulting graph
 is `Undirected`.
+The keyword argument can be used to consider the polyhedron modulo its lineality space.
```
```suggestion
If the polyhedron has lineality, then it has no vertices or bounded edges, so the `vertex_edge_graph` will be the empty graph.
In this case, the keyword argument can be used to consider the polyhedron modulo its lineality space.
```

benlorenz committed:
Update src/Combinatorics/Graphs/functions.jl

Co-authored-by: Lars Kastner <lkastner@users.noreply.github.com>

benlorenz commented:
I have changed this code to use (and rename) the existing `dualgraph` and `edgegraph` functions. I have kept aliases for the old names.
And I have adjusted them to deal with unbounded polyhedra:
- The vertex-edge-graph only considers proper vertices (unless using the `modulo_lineality` flag), i.e. it represents the bounded part of the polyhedron.
- The dual-graph corresponds to all (possibly unbounded) facets except for the facet at infinity. And facets that intersect only at infinity are not connected by an edge in the graph.

micjoswig commented:
> I have changed this code to use (and rename) the existing `dualgraph` and `edgegraph` functions. I have kept aliases for the old names. And I have adjusted them to deal with unbounded polyhedra:

Sounds perfect.  Thanks-

lgoettgens commented on src/aliases.jl:
```diff
@@ -48,3 +50,7 @@ function number_of_weak_compositions end
 @alias n_rows nrows
 @alias n_variables nvars
+
+# for backwards compatibility
+@alias dualgraph dual_graph
+@alias edgegraph vertex_edge_graph
```
Shouldn't these rather be `@deprecate`? Anyway, if they should be phased out they should be in `deprecations.jl` and not `aliases.jl`. `aliases.jl` only contains the aliases that we plan to keep forever

lkastner replied:
We discussed this internally and this is the result.
